integration_test.go

```diff
@@ -3848,3 +3848,64 @@ func TestTentacleCertificateResource(t *testing.T) {
 		return nil
 	})
 }
+
+func TestKubernetesAgentWorkerResource(t *testing.T) {
+	testFramework := test.OctopusContainerTest{
+		CustomEnvironment: map[string]string{
+			"OCTOPUS__FeatureToggles__KubernetesAgentAsWorkerFeatureToggle": "true",
+		},
+	}
+
+	testFramework.ArrangeTest(t, func(t *testing.T, container *test.OctopusContainer, spaceClient *client.Client) error {
+		// Act
+		newSpaceId, err := testFramework.Act(t, container, "./terraform", "58-kubernetesagentworker", []string{})
+
+		if err != nil {
+			return err
+		}
+
+		// Assert
+		client, err := octoclient.CreateClient(container.URI, newSpaceId, test.ApiKey)
+		query := machines.WorkersQuery{
+			CommunicationStyles: []string{"KubernetesTentacle"},
+			Skip:                0,
+			Take:                3,
+		}
+
+		resources, err := client.Workers.Get(query)
+		if err != nil {
+			return err
+		}
+
+		if len(resources.Items) != 2 {
+			t.Fatalf("Space must have two workers (both KubernetesTentacles), instead found %v", resources.Items)
+		}
+
+		optionalAgentName := "minimum-agent"
```

Comment:
`minimalAgent_` is a better variable name since the other agent has the name "agent-with-optionals" and it can be confusing